src/query/models/tag.go

```diff
@@ -145,17 +146,46 @@ func (m Matchers) ToTags() (Tags, error) {
 	return Normalize(tags), nil
 }
 
-// ID returns a string representation of the tags
-func (t Tags) ID() string {
-	b := make([]byte, 0, len(t))
+// StringID returns a string representation of the tags
+func (t Tags) StringID() string {
```

Comment:
why do you need to change the method name?

Reply:
because I added a BytesID(), so seems good to distinguish the two methods so the caller is forced to think about what they are doing

Reply:
fair. The goish (go-y?) way of doing this is to call WritesBytesID something like `MarshalTo()/MarshalInto()` 

maybe call the two methods `ID()` and `IDMarshalTo()`?